src/pybamm/solvers/c_solvers/idaklu/common.hpp

```diff
@@ -31,8 +31,8 @@
 #include <pybind11/stl.h>
 
 namespace py = pybind11;
-using np_array = py::array_t<realtype>;
-using np_array_dense = py::array_t<realtype, py::array::c_style | py::array::forcecast>;
+// note: we rely on c_style ordering for numpy arrays so don't change this!
```

Comment:
This is outside the scope of this PR, but we should be more careful about this in pybamm. We transpose the `y` when we store it with the python `Solution` object, so it's f-contiguous there. Not a big deal, but it'd make life simpler to keep it consistent.

Reply:
Agreed, I had to write the code assuming a particular ordering hence the specification as `c_style`. If an f_style ordered array is passed in then it gets converted to c_style by pybind11 (according to their docs: https://pybind11.readthedocs.io/en/stable/advanced/pycpp/numpy.html). This isn't ideal since it would involve a copy, but I wasn't sure about a better way of doing it. Let me know if you have any ideas

Reply:
Interesting, I didn't realize pybind11 made a copy here. IMO, I think we should stick with `c_style` and make python-side changes to `y` to avoid the transpose and change its ordering from `y[time, var]` to `y[var, time]`